lib/pbench/test/functional/server/test_user.py

```diff
@@ -0,0 +1,111 @@
+from http import HTTPStatus
+
+import pytest
+from requests import HTTPError
+
+from pbench.client import Pbench
+
+
+class TestUser:
+    """
+    Pbench Server functional tests for basic "user" operations.
+
+    NOTE: The functional test system is intended to run on a pristine
+    containerized Pbench Server. We could check for some pre-conditions if we
+    had a known Administrative user (e.g., admin/admin).
+    """
+
+    USERNAME = "functional"
+    PASSWORD = "tests"
+
+    def test_login_fail(self, pbench: Pbench):
+        """
+        Trying to log in to a non-existent user should fail.
+
+        NOTE: This will fail if "nouser" exists.
+        """
+        with pytest.raises(HTTPError) as e:
+            pbench.login("nouser", "nopassword")
+        assert (
+            str(e.value)
+            == "401 Client Error: UNAUTHORIZED for url: http://10.1.170.201/api/v1/login"
+        )
+
+    def test_register(self, pbench: Pbench):
+        """
+        Test that we can register a new user.
+
+        NOTE: This will fail if the username already exists.
```

Comment:
Should we have a test case which confirms the assertion in that NOTE?

Reply:
Yup; missed that one.